### PR TITLE
Webtorrent source build

### PR DESCRIPTION
**Pull request by bendlas**
###### Motivation for this change

Build more packages from source and re-use nixos' built electron.

Also version updates.

###### Things done

- Renamed `webtorrent_desktop` -> `webtorrent_desktop_bin` and removed hydra build
- Re-Initialized `webtorrent_desktop` as a source build with auto-updater

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @flokli

### CODE REVIEW

File: pkgs/applications/video/webtorrent_desktop/node-packages.nix [hunk 1]
```diff
@@ -0,0 +1,4386 @@
...
+      name = "load-ip-set";
+      packageName = "load-ip-set";
+      version = "2.1.0";
+      src = fetchurl {
+        url = "https://registry.npmjs.org/load-ip-set/-/load-ip-set-2.1.0.tgz";
+        sha512 = "taz7U6B+F7Zq90dfIKwqsB1CrFKelSEmMGC68OUqem8Cgd1QZygQBYb2Fk9i6muBSfH4xwF/Pjt4KKlAdOyWZw==";
+      };
+    };
+    "load-json-file-1.1.0" = {
+      name = "load-jso
```
Comment (bendlas) [inline pkgs/applications/video/webtorrent_desktop/node-packages.nix]: This is wrong. Let me check how to embed the fetch source here.

**Review by bendlas — COMMENTED** [review]
Doesn't build outside of my machine :-(

File: pkgs/applications/video/webtorrent_desktop/node-packages.nix (hunk 1 above)
Comment (bendlas) [reply pkgs/applications/video/webtorrent_desktop/node-packages.nix]: renders wront :-/
this is the line https://github.com/NixOS/nixpkgs/pull/74056/files/c8c8627fa36fddd5a3c2a106b5f2f0d5b592f0fb#diff-620905878edda2a966e61192c9004855R3777

**Comment by flokli** [comment]
This fails evaluation.

On top of that, I'm not sure if building electron apps from source is worth the effort - they tend to be quite strict in what version of electron they assume…

**Comment by bendlas** [comment]
webtorrent works fine in a range from electron 4-7, currently.

**Comment by bendlas** [comment]
the failure is because I used node2nix wrong. fixing ...

**Comment by flokli** [comment]
@bendlas it seems you never pushed the version fixing the evaluation.

Do you still want to pursue this?

**Comment by bendlas** [comment]
Pushed v0.21 https://github.com/NixOS/nixpkgs/commit/428b4353ce05f722d11bef317eb7a860689e27ca

I'd still like to take another stab at this, though ...

**Comment by bendlas** [comment]
Or not https://github.com/NixOS/nixpkgs/commit/c255b0fa45adfde495931fae3d44515fae11c715

**Comment by bendlas** [comment]
I got the source build fixed up and the program starts, but downloads don't work ... pushing the branch for reference, if somebody wants work on it, but closing for now. Pushed https://github.com/NixOS/nixpkgs/commit/9efcb767dac243702aca7be58bae3f0e49b8700e instead.

**Pull request closed without merging**